docs/spec/connection-handshake/interfaces.md

```diff
@@ -81,125 +85,58 @@ interface Proof {
 }
 ```
 
-##### New Handler version
-
-Now we redefine `connOpenTry` handler:
+##### New Handler (version 2)
 
+Now we redefine `connOpenTry` handler as follows:
 
 ```typescript
 function connOpenTry(
-  localConnEnd: ConnectionEnd,
-  remoteConnEnd: ConnectionEnd,
+  e: Connection,
   proofInit: Proof,
   proofConsensus: Proof) {
-    localConnId := remoteConnEnd.counterpartyConnectionIdentifier
-    abortTransactionUnless(validateConnectionIdentifier(localConnId))
+    abortTransactionUnless(validateConnectionIdentifier(e.remoteParty.connectionIdentifier))
     abortTransactionUnless(proofConsensus.height <= getCurrentHeight())
     expectedConsensusState = getConsensusState(proofConsensus.height)
 
-    // basic validation of connection ends
-    abortTransactionUnless(remoteConnEnd.state === INIT && localConnEnd.state === INIT)
-    abortTransactionUnless(remoteConnEnd.counterpartyPrefix === getCommitmentPrefix())
+    // basic validation of connection parties
+    abortTransactionUnless(e.state === INIT)
+    abortTransactionUnless(e.remoteParty.prefix === getCommitmentPrefix())
 
-    // verify the proofs
-    abortTransactionUnless(localConnEnd.verifyConnectionState(proofInit,  remoteConnEnd))
-    abortTransactionUnless(localConnEnd.verifyClientConsensusState(
+    abortTransactionUnless(e.remoteParty.verifyConnectionState(proofInit, e))
+    abortTransactionUnless(e.remoteParty.verifyClientConsensusState(
       proofConsensus, proofConsensus.height, expectedConsensusState))
-    previous = provableStore.get(connectionPath(localConnId))
+    previous = provableStore.get(connectionPath(e.remoteParty.connectionIdentifier))
     abortTransactionUnless(
       (previous === null) ||
-      (previous.state === INIT && previous === localConnEnd))
+      (previous === e))
+    identifier = e.remoteParty.connectionIdentifier
     state = TRYOPEN
-    provableStore.set(connectionPath(localConnEnd), connection)
-    addConnectionToClient(clientIdentifier, localConnEnd)
+    provableStore.set(connectionPath(identifier), connection)
+    addConnectionToClient(e.remoteParty.clientIdentifier, identifier)
 }
 ```
 
-### 3. Abstraction + no flipping
+Notice that this handler accesses predominantly the `remoteParty` fields. This happens because the relayer does not perform any "flipping" of parameters, so the perspective on this object is the one of the counterparty module. This can make it difficult to reason about, so below we discuss a version with flipping.
```

Comment:
Hmm, I am confused now. `e.localParty` is referring to the node running the handler above, no? And `e.remoteParty` to the other side. The relayer normally performs flipping.

Reply:
Alice and Bob are sending each other messages. Alice create connectionEnd and sends it to the Bob with a proof. As Alice initiates the connection, it chooses identifier for both side of the connection. Once Bob receives the message from Alice (connectionEnd with the proof), it verifies proof and creates the corresponding connectionEnd locally. I really don't understand why it is necessary talking about remoteParty view or flipping. Each party updates local state based on verified data it receives, and that's it.

Reply:
No, this version has no flipping. See below for the version with flipping.

Reply:
Ok, i am confused. In this version the flipping is done by the relayer (as it us currently) so when this handler receives `e` then `e.local` refers to its local state. Is this correct? If we agree can we be precise in what "flipping" means?

Reply:
ok, sorry i was looking at the wrong handler here...so let's show version3 before version2